app/src/main/java/org/mate/accessibility/check/screenbased/ErrorMessageScreenCheck.java

```diff
@@ -0,0 +1,81 @@
+package org.mate.accessibility.check.screenbased;
+
+import org.mate.MATE;
+import org.mate.accessibility.AccessibilityViolation;
+import org.mate.accessibility.AccessibilityViolationTypes;
+import org.mate.state.IScreenState;
+import org.mate.ui.Widget;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class ErrorMessageScreenCheck implements IScreenAccessibilityCheck{
+
+    private List<IScreenState> visitedStates;
+
+    public ErrorMessageScreenCheck(){
+        visitedStates = new ArrayList<IScreenState>();
+    }
+
+
+    @Override
+    public AccessibilityViolation check(IScreenState state) {
+
+        IScreenState visitedState =stateHasBeenVisited(state);
+
+        if (visitedState!=null){
+            MATE.log("same state");
+            if (differentContext(state,visitedState)){
+                AccessibilityViolation violation = new AccessibilityViolation(AccessibilityViolationTypes.ERROR_MESSAGE,state,"Content descriptions/hints do not changed after showing error message");
+                violation.setWarning(true);
+                return violation;
+            }
+        }
+        else{
+            MATE.log("different state");
+            visitedStates.add(state);
+        }
+
+        return null;
+    }
+
+    private boolean differentContext(IScreenState state, IScreenState visitedState) {
+
+        List<Widget> thisWidgets = state.getWidgets();
+        List<Widget> otherWidgets = visitedState.getWidgets();
+
+        boolean found = false;
+        for (Widget wThis: thisWidgets){
+            //search by id
+            for (Widget wOther: otherWidgets){
+                if (wThis.getId().contains("imageview_search)"))
+                {
+                    MATE.log("this: "+wThis.getText()+ " " + wThis.getContentDesc() + " " + wThis.getHint()+" " + wThis.getErrorText());
+                    MATE.log("other: "+wOther.getText()+ " " + wOther.getContentDesc() + " " + wOther.getHint()+" " + wOther.getErrorText());
+                }
+                if (wThis.getId().equals(wOther.getId())){
+
+
+
+                    //if (!wThis.getErrorText().equals(wOther.getErrorText())){
+                        if (wOther.getContentDesc().equals(wThis.getContentDesc()) && wOther.getHint().equals(wThis.getHint())) {
+                            return true;
+                        }
+                    //}
+
+                }
+            }
+        }
+        return false;
+    }
+
+    private IScreenState stateHasBeenVisited(IScreenState currentScreenState) {
+        List<IScreenState> recordedScreenStates = visitedStates;
+        for (IScreenState recordedScreenState : recordedScreenStates) {
```

Comment:
Use `if(visitedStates.contains(currentScreenState))` instead of the loop

Reply:
need to retry the state from the list though